src/api/node/sse/client_core.rs

```diff
@@ -0,0 +1,133 @@
+use super::{
+    error::ClientError,
+    types::{BoxedEventStream, CoreCommand, EventType, Handler},
+    SseData,
+};
+use eventsource_stream::{Event, Eventsource};
+use futures::stream::TryStreamExt;
+use std::collections::HashMap;
+
+pub struct ClientCore {
+    pub url: String,
+    pub event_stream: Option<BoxedEventStream>,
+    pub next_handler_id: u64,
+    pub event_handlers:
+        HashMap<EventType, HashMap<u64, Box<dyn Fn(SseData) + Send + Sync + 'static>>>,
+    pub id_types: HashMap<u64, EventType>,
+    pub is_connected: bool,
```

Comment:
Is the reason for having `is_connected` that we wish to support disconnecting and reconnecting?

When might a `ClientCore` become disconnected?

Reply:
The primary reason for that is to not attempt reading events from the stream if the client has not been connected yet. See [here](https://github.com/casper-network/casper-rust-sdk/pull/3/files#diff-cbd035d9c5229a99b63f8cf608d89c8762a562c28cf2b663b98cdc145e41d885R115)

Reply:
Could we switch to a builder pattern to give the connected client a different type from a `ClientBuilder`?

Reply:
Sounds good to me.

Reply:
I will create a follow-up PR switching to builder pattern not delaying the merge of this PR. Thanks for the suggestion @Avi-D-coder